src/QsCompiler/Transformations/FunctorGeneration.cs

```diff
@@ -148,6 +152,125 @@ public override QsStatementKind OnConjugation(QsConjugation stm)
         }
     }
 
+    /// <summary>
+    /// Scope transformation that splits any nested operation calls into separate statements so
+    /// that they can be properly reversed. This is necessary to avoid out of order execution of the
+    /// automatically generated adjoint. It is safe to do because an adjointable operation must return
+    /// Unit, so any nested calls can be replaced by Unit and those calls moved to separate,
+    /// ordered statements.
+    /// </summary>
+    public class ExtractNestedOperationCalls
```

Comment:
nit: In order to reduce additions to the public API, one trick that I've been doing is wrapping the transformation class in a public static class, making the transformation class private and adding a public static method to the wrapper class that instantiates and uses the (now private) transformation class in only the way that you intend it to be used. This removes the TransformationState nested class from the public API and ensures that the transformation is only used as intended, reducing the API addition to a single static method.
See Example: https://github.com/microsoft/qsharp-compiler/blob/79503a2a3ce31e9beeb7c65fc4d24fdc0c2cbdf4/src/QsCompiler/Transformations/ClassicallyControlled.cs#L31

Reply:
In this case, these transformations are already only used in the static methods in CodeTransformations.cs, so I just updated the whole file to be internal and now we don't have to worry about them being part of the public API surface.

Reply:
Please check with @bettinaheim about making the other existing transformations in this file internal before merging. I've mentioned that in the other comment, so I'll close this one out and we can consolidate the conversation there.